_lzbench/lzbench.cpp

```diff
@@ -233,6 +233,20 @@ size_t common(uint8_t *p1, uint8_t *p2)
     return size;
 }
 
+/*
+ * Allocate a buffer of size bytes using malloc (or equivalent call returning a buffer
+ * that can be passed to free). Touches each page so that the each page is actually
+ * physically allocated and mapped into the process.
+ */
+void *alloc_and_touch(size_t size, bool must_zero) {
```

Comment:
Yeah I could work, although then in the `must_zero` case you have to re-memset it after the zero, and then the compiler is possibly smart enough to elide the first memset and again do the "calloc" transformation. Of even if compiler version X isn't smart enough, then X+1 may be. A final advantage to this approach is that's a bit more minimal (at runtime, not in the code) since it only touches 1 out of 4096 bytes, so it's generally faster than `memset`, especially for smaller buffers (which may be cached).